src/schema.ts

```diff
@@ -92,8 +92,8 @@ function sanitize(value: any): void {
  */
 export default function schema<
   TProperties extends Record<string, unknown>,
-  TDefaults extends Partial<TProperties>,
-  TOptions extends SchemaOptions<TProperties, TDefaults>
+  TOptions extends SchemaOptions<TProperties>,
+  TDefaults extends TOptions['defaults'] = object
```

Comment:
Changing the order of these generics is a breaking change. Is there any way we can keep their order here?

Also, could we not initialize the `TDefaults` with `object`?

Reply:
Since we're not accessing the default types from the options object rather than allowing the returned type to be a `Partial<TProperties>` I don't believe we'll be able to preserve the original order of generics, unfortunately. 

As for initializing with `object`, I realize that's not ideal. To give context, the reason we're using `object` rather than `Record<string, never>` to represent an empty object is that `Record<string, never>` does not satisfy the generic constraints of `Partial<TProperties>`, but we need to have the type result in an empty object for instances where there is no default object present in the options object.

We could instead return `unknown` by not initializing the generic value at all - although that would mean updating `model` to accept that default value as well as a few new conditionals in the utils around building a new document for insert. It also feels like a bad choice to return `unknown` rather than `object` because we are actually returning an empty object from `schema` when no default object is present.

This would also not solve the issue of generic order being changed to avoid a breaking change.